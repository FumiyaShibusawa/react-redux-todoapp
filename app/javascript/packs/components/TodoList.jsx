import * as React from "react"

class TodoList extends React.Component {
  showForm = (e) => {
    $(e.target.parentElement).hide();
    $('#todo_list_form').show();
  }
  hideForm = (e) => {
    e.preventDefault();
    $('#todo_list_form').hide();
    $('[data-add="show-todolist"]').show();
  }

  render(){
    let input;
    return (
      <div className="todolist_component">
        <h2>TodoLists</h2>
        <ul className="todolists">
            { this.props.todo_lists.map( (todo_list, i) => (
              <li
              key={ `${todo_list.name}_${todo_list._id["$oid"]}` }
              onClick={ () => { this.props.showTodos(i) } }
              >{ todo_list.name }
              <span
              className="menu-ellipsis"
              key={ `${todo_list.name}_${todo_list._id["$oid"]}` }
              onClick={ (e) => {
                e.stopPropagation();
                this.props.deleteTodoList(todo_list);
              } }>︙</span>
              <div className="todolist-menu" style={{ display: 'none' }}>
                <ul>
                  <li>edit</li>
                  <li>delete</li>
                </ul>
              </div>
              </li>
            ) ) }
        </ul>
        <div className="add-button">
          <div data-add="show-todolist">
            <div className="plus">+</div>
            <span className="add-button-text" onClick={ this.showForm }>add new todolist</span>
          </div>
          <form id="todo_list_form" style={{ display: 'none' }} onSubmit={ (e) => {
              if (input) {
                e.preventDefault();
                this.props.addTodoList(input.value);
                input.value = ""; // テキストボックス内の値をクリア
              }
            } }>
            <input className="text-box" type="text" ref= { node => { input = node } }/>
            <div className="button-cont">
              <button type="submit" value="add">add</button>
              <div className="cancel" data-add="cancel" onClick={ this.hideForm }>cancel</div>
            </div>
          </form>
        </div>
      </div>
    )
  }
}

export default TodoList
