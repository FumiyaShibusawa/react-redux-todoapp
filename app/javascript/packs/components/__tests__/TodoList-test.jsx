import * as React from 'react';
import TodoList from '../TodoList';
import renderer from 'react-test-renderer';
import $ from 'jquery';
global.$ = global.jQuery = $;

test('it renders correctly', () => {
  const todo_lists = [
    {
      _id: { $oid: "5b2d02e7f450fe5251b9eb9d" },
      user_id: { $oid: "5b2cf7fef450fe50d9d60871" },
      created_at: "2018-06-22T14:08:39.983Z",
      updated_at: "2018-06-22T14:08:39.983Z",
      name: "write code everyday",
      todos: []
    }
  ]
  const tree = renderer.create(
    <TodoList
      todo_lists={todo_lists}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});