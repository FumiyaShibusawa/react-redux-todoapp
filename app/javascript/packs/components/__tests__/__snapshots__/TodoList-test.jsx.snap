// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> should render without error 1`] = `
<div
  className="todolist_component"
>
  <h2>
    TodoLists
  </h2>
  <ul
    className="todolists"
  >
    <li
      onClick={[Function]}
    >
      <span
        className="list-element"
      >
        write code everyday
      </span>
      <span
        className="menu-ellipsis"
        onClick={[Function]}
      >
        ï¸™
      </span>
    </li>
  </ul>
  <div
    className="add-button"
  >
    <div
      data-add="show-todolist"
    >
      <div
        className="plus"
      >
        +
      </div>
      <span
        className="add-button-text"
        onClick={[Function]}
      >
        add new todolist
      </span>
    </div>
  </div>
</div>
`;
